src/utilities/hypre_utilities.c

```diff
@@ -30,6 +34,58 @@ hypre_multmod(HYPRE_Int a,
     return res;
 }
 
+/*--------------------------------------------------------------------------
+ * hypre_MergeOrderedArrays: merge two ordered arrays
+ *--------------------------------------------------------------------------*/
+
```

Comment:
There exists `hypre_union2` that seems to have the same functionality.

Reply:
Thanks for pointing this out, @liruipeng! `hypre_union2` works on arrays of `HYPRE_BigInt` and `hypre_MergeOrderedArrays` works on `HYPRE_Int`. Also, the second doesn't have the extra step of checking the map arrays, so I think we need both. But maybe a name change would help unifying things? Something like `hypre_MergeOrderedIntArray` and `hypre_MergeOrderedBigIntArray`.

Reply:
You can set `map==NULL` to avoid the mapping thing. That's not a problem I think, while the `BigInt` part is. The algorithm (merge sort) is the same. It is too bad we have to write the same thing twice just because the type difference.

Reply:
We actually have more implementations on this... How about `hypre_merge_sort`?

Reply:
Good point! Do you know if it removes duplicate entries?

Reply:
Not sure about whether it removes duplicates. I assume by duplicate entries you meant an entry that appears in both `arr1` and `arr2`, but `arr1` or `arr2` is not supposed to have "duplicate entries". (see the comments in `hypre_union2`).

Looking at `hypre_merge_sort`, it is more complicated since it is parallel. I think the name change @victorapm suggested is good enough, that is changing the current `hypre_union2` to `hypre_MergeOrderedBigIntArray`, with some interface changing maybe. The majority of these two functions seems to be the same.

Reply:
Also, it may be better put it in `hypre_merge_sort.c`

Reply:
Yes, that's what I meant for duplicate entries. So, `hypre_union2` can't be used. Actually, maybe it makes more sense to call `hypre_MergeOrderedArrays` as `hypre_UnifyOrderedIntArrays` and `hypre_union2` as `hypre_MergeOrderedBigIntArrays` since the first removes duplicate entries and the second doesn't.

Reply:
`hypre_union2` also removes duplicates. If you compare the code of these two, they are the same pretty much.

Reply:
Yes, you're correct. So, the header comments of `hypre_union2` are misleading. The first assumption says `no duplicates in arr1 and arr2`, but it clearly works also for the case when this doesn't hold

Reply:
> Yes, you're correct. So, the header comments of `hypre_union2` are misleading. The first assumption says `no duplicates in arr1 and arr2`, but it clearly works also for the case when this doesn't hold

OK. I think the confusion is about where the duplicates can/cannot exist. The assumption in `hypre_union2` means in `arr1` or `arr2` there is no entry that appears more than once, but an entry is allowed to appear in both `arr1` and `arr2`. At least this is the case where I used this function.

Reply:
Thanks for clarifying, Ruipeng! I updated this header comment to make it more clear and moved `hypre_MergeOrderedArrays` to `hypre_merge_sort.c` in 8e8849c. Still, I'm leaving two separate functions even if the algorithm is basically the same because they operate in different kinds of data